src/node_file.cc

```diff
@@ -95,11 +97,77 @@ class FSReqWrap: public ReqWrap<uv_fs_t> {
   DISALLOW_COPY_AND_ASSIGN(FSReqWrap);
 };
 
+class Path {
+  public:
+    explicit Path(v8::Isolate* isolate, v8::Local<v8::Value> value);
+
+    ~Path() {
+      if (str_ != str_st_)
+        free(str_);
+    }
+
+    char* operator*() {
+      return str_;
+    };
+
+    const char* operator*() const {
+      return str_;
+    };
+
+    size_t length() const {
+      return length_;
+    };
+
+    bool fail() const {
+      return fail_;
+    }
+
+  private:
+    bool fail_ = false;
+    size_t length_;
+    char* str_;
+    char str_st_[1024];
+};
+Path::Path(v8::Isolate* isolate, v8::Local<v8::Value> value)
+    : length_(0), str_(str_st_) {
+  if (value.IsEmpty())
+    return;
+
+  if (value->IsString()) {
+    v8::Local<v8::String> string = value->ToString(isolate);
+    if (string.IsEmpty())
+      return;
+
+    // Allocate enough space to include the null terminator
+    size_t len = StringBytes::StorageSize(isolate, string, UTF8) + 1;
+    if (len > sizeof(str_st_)) {
+      str_ = static_cast<char*>(malloc(len));
+      CHECK_NE(str_, nullptr);
+    }
+
+    const int flags =
+        v8::String::NO_NULL_TERMINATION | v8::String::REPLACE_INVALID_UTF8;
+    length_ = string->WriteUtf8(str_, len, 0, flags);
+    str_[length_] = '\0';
+  } else if (Buffer::HasInstance(value)) {
+    size_t len = Buffer::Length(value) + 1;
+    if (len > sizeof(str_st_)) {
+      str_ = static_cast<char*>(calloc(len, 1));
```

Comment:
think I had him do that. was being lazy with an example and called calloc instead of typing `str_[n] = '\0';`. since the string needs to be null-terminated.